manifests/init.pp

```diff
@@ -451,10 +451,12 @@
       $apm_footer_order = '06'
     }
 
-    concat::fragment{ 'datadog apm footer':
-      target  => '/etc/dd-agent/datadog.conf',
-      content => template('datadog_agent/datadog_apm_footer.conf.erb'),
-      order   => $apm_footer_order,
+    if $apm_env != '' {
```

Comment:
would it maybe make more sense to check for `$apm_enabled`?

Reply:
That makes sense but if apm_enabled == true but apm_env is not set then you would still get an error.

I could add a check for both so long as we think the user will know they need to set both variables to have it configured.